bson/son.py

```diff
@@ -176,11 +182,11 @@ def transform_value(value):
 
         return transform_value(dict(self))
 
-    def __deepcopy__(self, memo):
-        out = SON()
+    def __deepcopy__(self, memo: Dict[int, "SON[_Key, _Value]"]) -> "SON[_Key, _Value]":
+        out: SON[_Key, _Value] = SON()
         val_id = id(self)
         if val_id in memo:
-            return memo.get(val_id)
+            return memo.get(val_id)  # type: ignore
```

Comment:
We can make this type-safe and remove `# type: ignore` with `return memo[val_id]` instead of using .get().

Reply:
Oh, ha, duh, thanks